toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -1285,6 +1392,42 @@ func copyAdditionalFiles(installChroot *safechroot.Chroot, config configuration.
 	return
 }
 
+// cleanupRpmDatabase removes RPM database if the image does not require a package manager.
+// rootPrefix is prepended to the RPM database path - useful when RPM database resides in a chroot and cleanupRpmDatabase can't be called from within the chroot.
+// isRootFS should be set to true if the resulting image will be a rootfs (not a file)
+// packagesToInstall is a list of packages that will be installed on the image
+func cleanupRpmDatabase(rootPrefix string, isRootFS bool, packagesToInstall []string) {
+	if !isRootFS {
+		logger.Log.Debug("Processing a non-rootfs. Skipping RPM database cleanup.")
+		return
+	}
+
+	// If the image doesn't contain the package manager
+	// We can remove the RPM database files
+	rpmInChroot := false
+
+	for _, name := range []string{"rpm", "dnf", "tdnf", "yum"} {
```

Comment:
`packagesToInstall` may only contains top level dependencies. This means there are very legitimate instances where this routine would remove the rpm database from a system with a package manager. 

For example an image may request package `foo` to be installed, and `foo` has a runtime requirement on `rpm`. 

Suggestion: I think the best way to check here would be scan the chroot after all packages have been installed for a package manager. Perhaps you could use [exec.LookPath](https://golang.org/pkg/os/exec/#LookPath).

Reply:
Perhaps more preferably, removing the rpm database could be an option in the image configuration file to make it a more explicit decision instead of hidden behavior.

Reply:
Re-opening. If the option is controlled by a flag I think it should be either on-or-off based on the user's setting, not automatic detection. From my first comment this auto detection has scenarios that it will miss, leading to non-desired results in certain instances.

I think the best option here is to make the image config option "RemoveRpmDatabase", and always perform the database cleanup if set.